dglgo/dglgo/apply_pipeline/graphpred/graphpred.jinja-py

```diff
@@ -0,0 +1,77 @@
+import torch
+import os
+import csv
+
+from tqdm import tqdm
+from dgl.data import AsGraphPredDataset
+from dgl.dataloading import GraphDataLoader
+{{ data_import_code }}
+
+{{ model_code }}
+
+def infer(device, loader, model):
+    model = model.to(device)
+    model.eval()
+    all_pred = []
+
+    with torch.no_grad():
+        for _, (g, labels) in enumerate(tqdm(loader, desc="Iteration")):
+            g = g.to(device)
+            node_feat = g.ndata['feat']
+            edge_feat = g.edata['feat']
+            pred = model(g, node_feat, edge_feat)
+            pred = (pred.sigmoid() >= 0.5).long()
```

Comment:
Look like we only support binary classification here. Probably should add a check in the training script.

Reply:
We only support binary classification for both training and inference. We might want to make it clear in the doc. However, there isn't much that can be done in the scripts.

Reply:
Besides doc update, I suggest also have a check of the label shape in the training script to make sure it is binary classification.

Reply:
`label shape` cannot tell you about that. For multi-label binary classification, the shape is `(N, T)`, where `T` is the number of tasks. For multi-class classification, the shape is `(N, C)`, where `C` is the number of classes.

Reply:
Got it. Then let's make it clear that this pipeline is for binary classification (in doc and in commandline help message as well).

Reply:
updated